### PR TITLE
Update the RID target from lower Linux to the latest one

### PR DESCRIPTION
**Pull request by v-wuzhai**

### CODE REVIEW

File: .vsts-ci.yml [hunk 1]
```diff
@@ -126,18 +126,18 @@ stages:
 
     - template: /eng/build.yml
       parameters:
-        agentOs: Ubuntu_18_04
+        agentOs: Ubuntu_20_04
         pool:
           ${{ if eq(variables['System.TeamProject'], 'public') }}:
             name: NetCore1ESPool-Svc-Public
-            demands: ImageOverride -equals Build.Ubuntu.1804.Amd64.Open
+            demands: ImageOverride -equals 1es-ubuntu-2004-open
```
Comment (marcpopMSFT) [inline .vsts-ci.yml:133]: @MattGal can you review the yml changes here as I'm less familiar with which images we should or shouldn't be using.

File: .vsts-ci.yml [hunk 2]
```diff
@@ -126,18 +126,18 @@ stages:
...
           ${{ if ne(variables['System.TeamProject'], 'public') }}:
             name: NetCore1ESPool-Svc-Internal
-            demands: ImageOverride -equals Build.Ubuntu.1804.Amd64
+            demands: ImageOverride -equals 1es-ubuntu-2004
         ${{ if eq(variables['System.TeamProject'], 'public') }}:
-          helixTargetQueue: Ubuntu.1804.Amd64.Open
+          helixTargetQueue: 'ubuntu.2004.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-20.04-helix-amd64-20220502145738-4b2e4c2'
```
Comment (MattGal) [inline .vsts-ci.yml:138]: Running in "docker" mode incurs a time penalty (initial fetch of image & the cost of being a nested VM).  Is there an intentional reason why you're running 20.04 inside 20.04?

File: .vsts-ci.yml [hunk 3]
```diff
@@ -126,18 +126,18 @@ stages:
...
         ${{ if ne(variables['System.TeamProject'], 'public') }}:
-          helixTargetQueue: Ubuntu.1804.Amd64
+          helixTargetQueue: Ubuntu.2004.Amd64
```
Comment (MattGal) [inline .vsts-ci.yml:140]: You're also not using the docker tag here; whichever you decide you need should probably be in both places

**Review by MattGal — APPROVED** [review]
I think you have no reason to be using docker helix target queue stuff here, and should definitely do the same thing in internal/public just for continuity, the changes look valid otherwise.

**Comment by marcpopMSFT** [comment]
@v-wuzhai can you resolve the merge conflicts?  Thanks.

**Comment by v-wuzhai** [comment]
Done

**Review by marcpopMSFT — APPROVED** [review]

**Pull request merged**